.github/workflows/build.yml

```diff
@@ -25,6 +29,57 @@ jobs:
       - name: Build and Test
         run: make build-and-test
 
+  run-integration-tests:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python: [ "3.10" ]
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v3
+        with:
+          python-version: "${{ matrix.python }}"
+      - name: Test Python version
+        run: |
+          installed="$(python --version)"
+          expected="${{ matrix.python }}"
+          echo $installed
+          [[ $installed =~ "Python ${expected}" ]] && echo "Configured Python" || (echo "Failed to configure Python" && exit 1)
+      - name: Install required os level applications
+        run: |
+          sudo apt update -y
+          sudo apt install -y curl git nodejs
+          sudo apt -y install ca-certificates curl gnupg lsb-release
+          sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+          sudo echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+            $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+          sudo apt update
+          sudo apt -y install docker-ce docker-ce-cli containerd.io
+      - name: Setup docker compose
+        run: |
+           sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+           sudo chmod +x /usr/local/bin/docker-compose
+           sudo ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+           docker-compose --version
+      - name: Check cache for Docker layers
+        uses: satackey/action-docker-layer-caching@v0.0.11
+        # Ignore the failure of a step and avoid terminating the job:
+        continue-on-error: true
+        with:
+          key: docker-layer-caching-${{ github.workflow }}-${{ hashFiles(env.SANDBOX_CONFIG, env.SANDBOX_GENESIS) }}-{hash}
```

Comment:
@tzaffi A few questions here:

* Is the provided key still accurate?
  * https://github.com/algorand/pyteal/runs/6164428544?check_suite_focus=true suggests to me that _only_ `makerxau/algorand-sandbox-dev:latest` is cached.  So, it feels to me like the following changes can be made:
    * Remove `env.SANDBOX_CONFIG` and `env.SANDBOX_GENESIS`.
    * Remove `key` override to use the default name.
* Since the PR refers to a `latest` image, does github caching provide the expected behavior?  
  * For `makerxau/algorand-sandbox-dev`, when a new image version appears as `latest`, will CI pick up the change?
    * I don't know off-hand, but I'm guessing _no_. 
  * I see https://github.com/MakerXStudio/algorand-sandbox-dev does _not_ provide tags.  
  * If we expect new image versions to be picked up and provided the build time remains tolerable, then I propose removing the caching.

Reply:
Last I checked, they only put one docker image up. One hesitation to this approach is the possibility of support being dropped and the docker image going stale. I have [forked the repo](https://github.com/tzaffi/algorand-sandbox-dev) and it has an MIT license which means we can build the image against a newer `go-algorand` and push to our own dockerhub. I've verified that rebuilding the image works as expected and I was able to pass PyTEAL integration test on it. I've even created and uploaded that image (`tzaffi/algod:24apr2022`)

Since caching isn't really any faster than not, I'll remove it altogether. If build times become less tolerable, we can re-introduce.

Action items:
- [ ]  remove `env.SANDBOX_*`
- [ ]  remove caching
- [ ]  test against `tzaffi/algod:24apr2022`

Reply:
@tzaffi Few thoughts:
* Thanks for removing caching via https://github.com/algorand/pyteal/pull/249/commits/d56772f3d330dcf6f7df9a276c5cd3ccb8b5242d.
* For `tzaffi/algod:24apr2022` (from https://github.com/algorand/pyteal/commit/50f59c93bc1b16b32f6cfa718e573dd0046e61e2), how will other maintainers trace the image's lineage back to https://github.com/tzaffi/algorand-sandbox-dev?
  * I agree forking the repo to maintain our own image is a good move.  Though I'm wondering if we can stay with the prior image and ticket the desire to move to a self-managed image.

Reply:
That approach works too. I'll create a ticket for that.

Reply:
Here's the ticket in `algorand-internal` https://github.com/algorand/go-algorand-internal/issues/2004

Reply:
@tzaffi Thanks for taking the time to ticket https://github.com/algorand/go-algorand-internal/issues/2004.  Resolving.